src/components/ShopingCart/ShopingCart.jsx

```diff
@@ -0,0 +1,198 @@
+    category: "Programming",
+    numOfVideos: 150,
+    numOfSections: 8,
+    numOfLectures: 4,
+    price: 50.0,
+    rating: 4.5,
+  },
+  {
+    image: course1,
+    author: "Adel nsiem",
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfLectures: 4,
+    numOfSections: 8,
+    price: 50.0,
+    rating: 4.5,
+  },
+];
+function ShopingCart() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const productsPerPage = 4;
+  const getCurrentPage = (pageIndex, productsPerPage) => {
+    setCurrentPage(pageIndex);
+  };
+  const indexOfLastCourse = currentPage * productsPerPage;
+  const indexOfFirstCourse = indexOfLastCourse - productsPerPage;
+  const currentProducts = ShopingCartCourses.slice(
+    indexOfFirstCourse,
+    indexOfLastCourse
+  );
+  return (
+    <>
+      {/* Header Of Shopping Cart */}
+      <div className="flex justify-center place-content-center bg-stone-700 bg-opacity-8 p-[1.5rem] w-full">
+        <h1 className="text-white text-[1rem] md:text-[1.5rem] xl:text-[2rem] font-semibold tracking-wider">
+          Shopping Cart <FontAwesomeIcon icon={faShoppingCart} />
+        </h1>
+      </div>
+      {/* render courses */}
+      <div className="flex flex-col md:flex-row mx-4 mb-8">
+        <div className="ml-0 md:ml-10 xl:ml-32 mt-20 w-full md:w-7/12">
+          <div className="mb-3">
+            <h1 className="text-2xl font-bold">Courses</h1>
+          </div>
+          {ShopingCartCourses.slice(
+            (currentPage - 1) * productsPerPage,
+            currentPage * productsPerPage
+          ).map((item) => {
+            return (
+              <div
+                key={item}
+                className="mb-2 relative flex pl-[1.rem] pr-5 pt-4 pb-3 bg-white rounded-[0.3rem] border-b border-black border-opacity-60"
+              >
+                <div>
+                  <img
+                    src={item.image}
+                    className="ml-2 w-24 md:w-28 h-24"
+                    alt={item.title}
+                    loading="lazy"
+                  />
+                </div>
+                <div className="ml-5">
+                  <div className="flex ">
+                    <Link className="text-blue-900 hover:text-blue-950 transition duration-300 ease-in-out text-sm font-semibold tracking-tight xl:text-lg">
+                      {item.title}
+                    </Link>
+                    <h2 className="absolute right-1 text-black text-base font-semibold tracking-tight ">
+                      {item.price} $
+                    </h2>
+                  </div>
+                  <h2 className="text-black text-opacity-40 text-base font-medium tracking-tight">
+                    by {item.author}
+                  </h2>
+                  <span className="text-black text-lg ">{item.rating}</span>
+                  {Array.from({ length: item.rating }, (_, index) => (
```

Comment:
The "length" property can't be float data type, it should be an integer, whatever we need to set the length to 5 (5 stars), Also we need to remove the span which represents the "rate" number, and try to color the 5 stars depending on the rate number, please fix it.

Reply:
ok i try to do this .